sys/net/gnrc/network_layer/ipv6/nib/nib_pl.c

```diff
@@ -69,7 +69,7 @@ int gnrc_ipv6_nib_pl_set(unsigned iface,
         (ipv6_addr_match_prefix(&netif->ipv6.addrs[idx], pfx) >= pfx_len)) {
         dst->flags |= _PFX_ON_LINK;
     }
-    if (netif->ipv6.aac_mode == GNRC_NETIF_AAC_AUTO) {
+    if (netif->ipv6.aac_mode |= GNRC_NETIF_AAC_FLAGS_DHCP) {
```

Comment:
This is not how flag checking works.

```suggestion
    if (netif->ipv6.aac_mode & GNRC_NETIF_AAC_FLAGS_DHCP) {
```